sql/migrate/migrate.go

```diff
@@ -127,303 +123,140 @@ func Schema(s *schema.Schema) StateReader {
 }
 
 type (
-	// Dir represents a versioned migration directory.
-	Dir struct {
-		fs        fs.FS
-		conn      Driver
-		pattern   string
-		templates []struct{ N, T *template.Template }
-	}
+	// FS describes the methods needed to work with the Planner.
+	FS interface {
+		// List all present migration files. Usually *.sql files.
+		List() ([]string, error)
 
-	// DirOption allows configuring the Dir
-	// using functional options.
-	DirOption func(*Dir) error
-)
+		// Read the contents of a file by name.
+		Read(name string) ([]byte, error)
```

Comment:
Can we use the standard [io.ReasDirFS](https://pkg.go.dev/io/fs#ReadDirFS) here?